advanced_source/visualizing_gradients_tutorial.py

```diff
@@ -0,0 +1,593 @@
+"""
+Visualizing Gradients
+=====================
+
+**Author:** `Justin Silver <https://github.com/j-silv>`__
+
+When training neural networks with PyTorch, it’s possible to ignore some
+of the library’s internal mechanisms. For example, running
+backpropagation requires a simple call to ``backward()``. This tutorial
+dives into how those gradients are calculated and stored in two
+different kinds of PyTorch tensors: leaf vs. non-leaf. It will also
+cover how we can extract and visualize gradients at any layer in the
+network’s computational graph. By inspecting how information flows from
+the end of the network to the parameters we want to optimize, we can
+debug issues that occur during training such as `vanishing or exploding
+gradients <https://arxiv.org/abs/1211.5063>`__.
+
+By the end of this tutorial, you will be able to:
```

Comment:
This section can be removed if you explicitly mention what we want to highlight in the previous section.